### PR TITLE
update scipy 1.13.1

### PR DESCRIPTION
**Pull request by wmboon**
The test now assembles the known sparse arrays and compares only after changing to canonical format.

### CODE REVIEW

File: tests/unit/test_vecvlagrange1.py [hunk 1]
```diff
@@ -20,28 +20,32 @@ def setup(self, sd):
...
         stiff.data[np.abs(stiff.data) < 1e-10] = 0
-        stiff.eliminate_zeros()
+
+        for X in [div, symgrad, div_div, pen, symgrad_symgrad, diff, stiff]:
+            self.make_canonical(X)
 
         return M, div, symgrad, div_div, symgrad_symgrad, pen, diff, stiff, discr
 
+    def make_canonical(self, A: sps.sparray):
+        A.sum_duplicates()
```
Comment (alessiofumagalli) [inline tests/unit/test_vecvlagrange1.py:40]: do you think we need to do the sum_duplicates also for the other functions in the discretization classes?

File: tests/unit/test_vecvlagrange1.py [hunk 2]
```diff
@@ -20,28 +20,32 @@ def setup(self, sd):
...
+        A.eliminate_zeros()
+        A.sort_indices()
+
+    def assemble_known(self, data, indices, indptr, shape=None):
+        A = sps.csc_array((data, indices, indptr), shape=shape)
```
Comment (alessiofumagalli) [inline tests/unit/test_vecvlagrange1.py:45]: did we move completely to array?

**Review by alessiofumagalli — APPROVED** [review]
the pr looks good, I just have two general questions that are given in the specific lines of the file

File: tests/unit/test_vecvlagrange1.py (hunk 1 above)
Comment (wmboon) [reply tests/unit/test_vecvlagrange1.py:40]: I don't know. It seems to be only a problem here and it is not clear to me why. I would suggest we only do it if/when it becomes necessary.

We may need to look for a different way to compare sparse arrays

File: tests/unit/test_vecvlagrange1.py (hunk 2 above)
Comment (wmboon) [reply tests/unit/test_vecvlagrange1.py:45]: Not yet but we will

**Pull request merged**